### PR TITLE
feat(schemas): add sie2 schema

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Update schema to fit SIE v2 design: https://www.figma.com/file/NJ7kpxondnIRMt3a933x4E/%F0%9F%8F%96%EF%B8%8F-%5BAC%5D-SIE-v2?node-id=215%3A7942

PLEASE NOTICE THAT this is merging to `feature/sie-v2` not `master`, so the project is not completed.

## Testing

can not be tested now, wait for the incoming PRs.

### CODE REVIEW

File: packages/core/src/__mocks__/sign-in-experience.ts [hunk 1]
```diff
@@ -30,6 +32,15 @@ export const mockSignInExperience: SignInExperience = {
...
   },
+  signUp: {
+    identifier: SignUpIdentifier.Username,
+    password: true,
+    verify: false,
+  },
+  signIn: {
+    methods: [{ identifier: SignInIdentifier.Username, password: true, verificationCode: false }],
+    isPasswordPrimary: true,
+  },
```
Comment (gao-sun) [inline packages/core/src/__mocks__/sign-in-experience.ts:43]: should submit with the migration script?

Comment (wangsijie) [reply packages/core/src/__mocks__/sign-in-experience.ts:43]: Well, will create another PR merging to [feature/sie-v2](https://github.com/logto-io/logto/tree/feature/sie-v2)

File: packages/schemas/src/foundations/jsonb-types.ts [hunk 2]
```diff
@@ -130,6 +130,41 @@ export const signInMethodsGuard = z.object({
...
+
+export const signInGuard = z.object({
+  methods: z
+    .object({
+      identifier: z.nativeEnum(SignInIdentifier),
+      password: z.boolean(),
+      verificationCode: z.boolean(),
+    })
+    .array(),
+  isPasswordPrimary: z.boolean(),
```
Comment (darcyYe) [inline packages/schemas/src/foundations/jsonb-types.ts:163]: try to confirm: the priority of username+password matters, and the order between email/sms passwordless is not important?

File: packages/schemas/tables/sign_in_experiences.sql [hunk 3]
```diff
@@ -7,7 +7,10 @@ create table sign_in_experiences (
   language_info jsonb /* @use LanguageInfo */ not null,
   terms_of_use jsonb /* @use TermsOfUse */ not null,
   sign_in_methods jsonb /* @use SignInMethods */ not null,
```
Comment (darcyYe) [inline packages/schemas/tables/sign_in_experiences.sql:9]: Sie v2 depends on `sign_in` and `sign_up` but not `sign_in_methods` in the future?
Are we going to make sie v1 compatible? Or will we drop `sign_in_methods` in another PR

File: packages/schemas/src/foundations/jsonb-types.ts (hunk 2 above)
Comment (wangsijie) [reply packages/schemas/src/foundations/jsonb-types.ts:163]: is the order between password and verification code

File: packages/schemas/tables/sign_in_experiences.sql (hunk 3 above)
Comment (wangsijie) [reply packages/schemas/tables/sign_in_experiences.sql:9]: can not drop `sign_in_methods` until all jobs are done.

**Review by darcyYe — APPROVED** [review]

**Review by simeng-li — APPROVED** [review]
LGTM

**Pull request merged**